examples/getting_started/web_search.ipynb

```diff
@@ -0,0 +1,255 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "def web_search(query: str) -> str:\n",
+        "    \"\"\"Perform a web search and return the results.\n",
+        "\n",
+        "    This is a simplified implementation. In a real-world scenario,\n",
+        "    you would integrate with a proper search engine API.\n",
+        "    \"\"\"\n",
+        "    # Placeholder implementation\n",
+        "    return f\"Search results for: {query}\"\n",
+        "\n",
+        "\n",
+        "def parse_webpage(url: str) -> str:\n",
+        "    \"\"\"Parse the content of a webpage.\"\"\"\n",
+        "    try:\n",
+        "        response = requests.get(url)\n",
+        "        soup = BeautifulSoup(response.content, \"html.parser\")\n",
+        "        return \"\\n\".join([p.text for p in soup.find_all(\"p\")])\n",
+        "    except Exception as e:\n",
+        "        return f\"Error parsing webpage: {str(e)}\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## 4. Building the Web Search Agent <a name=\"building\"></a>\n",
+        "\n",
+        "Now that we have our tools ready, let's build the WebSearchAgent class. This class will use the Mirascope framework to create an AI agent that can interact with users, perform web searches, and provide informative responses."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "MAX_RESULTS = 10\n",
+        "\n",
+        "\n",
+        "class WebSearchAgent(BaseModel):\n",
+        "    history: list[ChatCompletionMessageParam] = []\n",
+        "    max_results: int = MAX_RESULTS\n",
+        "\n",
+        "    @openai.call(\"gpt-4o-mini\", stream=True, tools=[parse_webpage, web_search])\n",
+        "    @prompt_template(\"\"\"\n",
+        "        SYSTEM:\n",
+        "        Your task is to answer a user's query.\n",
+        "\n",
+        "        You have access to the following tools:\n",
+        "        - `web_search`: Search the web for information (limited to {self.max_results} results).\n",
+        "        - `parse_webpage`: Parse the content of a webpage.\n",
+        "\n",
+        "        When calling the `web_search` tool, the `body` is simply the body of the search\n",
+        "        result. You MUST then call the `parse_webpage` tool to get the actual content\n",
+        "        of the webpage. It is up to you to determine which search results to parse.\n",
+        "\n",
+        "        Once you have gathered all of the information you need, generate your response,\n",
+        "        which should strike the right balance between brevity and completeness. You answer\n",
+        "        to the user's query should be concise yet comprehensive.\n",
```

Comment:
nit: it's useful to include the current date as this will often provide better results (particularly for requests that ask about "recent" stuff where the LLM might use the date of its training cutoff rather than the actual current date because it doesn't know the current date unless we tell it).